GitCommands/Settings/AppSettings.cs

```diff
@@ -276,6 +276,67 @@ public static bool AlwaysShowAllCommands
             set => WriteBoolRegKey("AlwaysShowAllCommands", value);
         }
 
+        public static bool EnableExplorerIntegration
+        {
+            // TODO write getter - check if dll is registered
+            // get => ReadBoolRegKey("AlwaysShowAllCommands", false);
+            set => HandleShellIntegration(null, null, value);
+        }
+
+        private static void HandleShellIntegration(object sender = null, EventArgs e = null, bool register = false)
```

Comment:
Please move out to own file, call it something like `ShellExtensionManager`:
```cs
public sealed ShellExtensionManager
{
	/// <summary>
	///  Check whether the shell extensions is registered.
	/// </summary>
	/// <returns>
	///  <see langword="true"/> if the shell extensions is registered; otherwise <see langword="false"/>.
	/// </returns>
	public bool CheckRegistered();

	/// <summary>
	///  Registers the shell extensions. Throws an exception, if registration process failed.
	/// </summary>
	public void Register([NotNull] string extensionLocation);

	/// <summary>
	///  Unregisters the shell extensions. Throws an exception, if registration process failed.
	/// </summary>
	public void Unregister([NotNull] string extensionLocation);
}
```

NB: don't make it static, to reduce risks of race conditions while testing. The implementation should be stateless (as far as our code is concerned), so there is no issue with creating multiple instances of the class.
We can later replace is with MEF DI.

Furthermore, I don't like the functionality of registering/unregestering being placed in AppSettings. It is not a settings, and it is unrelated to the app.
If there is an error while registering or unregistering the shell extensions there is no way to inform a user.
Please move checks and register/unregister to the settings page, where it belongs.